### PR TITLE
feat: solo home directory cleanup

### LINKED ISSUE #1560: delete specific SOLO_CACHE_DIR components as part of X command/sub-commands
Command/sub-commands to delete SOLO_CACHE_DIR components:
- rename `--dev` to `--debug`
- if `--debug` is NOT enabled: delete the SOLO_CACHE_DIR keys and **/keys/* at the end of the `solo network deploy` command (or after we have deployed them to the K8s cluster(s))

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes the following:
- Ensures that no files in SOLO_HOME have higher file permissions than 750.
- Deprecates the `--dev` flag and introduces a new, more intuitively named `--debug` flag.
- Keys are no longer stored on disk in SOLO_HOME.
- Removes the `~/.solo/cache/last-one-shot-deployment.txt` file, as it is no longer needed. It used to store the name of the deployment, and that was useful when it was randomly generated. But since it now defaults to the constant `one-shot`, the file is not needed.

### Related Issues

* Closes #1560 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/core/platform-installer.ts [hunk 1]
```diff
@@ -512,4 +516,57 @@ export class PlatformInstaller {
...
+  /**
+   * Resolve the base64-encoded contents of a key file needed to (re)build a Kubernetes secret.
+   *
+   * Prefers the on-disk copy in the keys directory; when that is absent (e.g. the keys were removed
+   * after `network deploy` ran without --debug) it falls back to the value already stored in the
+   * owning node's Kubernetes secret, which keeps each entry base64-encoded under its file basename.
+   * @param keysDirectory - the directory that may contain the key file
+   * @param fileName - the key file basename (also the secret data key)
+   * @param readSecretData - reads the fallback secret's data map for the node that owns this file
+   */
```
Comment (jan-milenkov) [inline src/core/platform-installer.ts]: leftover comment

**Comment by jan-milenkov** [comment]
Except for the leftover comment, the rest seems ok with me, @jeromy-cannon  i think i'll even have to delay the flowchart until those changes are merged

File: src/core/platform-installer.ts (hunk 1 above)
Comment (Ivo-Yankov) [reply src/core/platform-installer.ts]: removed

**Review by JeffreyDallas — COMMENTED** [review]
"Deprecates the --dev flag and introduces a new, more intuitively named --debug flag" this alone could be a separate PR to lower the size of this PR.

**Comment by JeffreyDallas** [comment]
@deshmukhpranali Please be aware of this change to update solo-docs later "~/.solo/cache/last-one-shot-deployment.txt " will be removed

**Review by JeffreyDallas — COMMENTED** [review]
Remove "last-one-shot-deployment.txt" also could be a separate PR to make this PR more focus and easy to review

**Review by jan-milenkov — APPROVED** [review]

**Comment by Ivo-Yankov** [comment]
@JeffreyDallas I'm sorry for the size of this PR. I opened to PRs in solo-docs to reflect the changes here:
- https://github.com/hiero-ledger/solo-docs/pull/271
- https://github.com/hiero-ledger/solo-docs/pull/272
